api/integration.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+// IntegrationAccount (https://discord.com/developers/docs/resources/guild#integration-account-object)
+type IntegrationAccount struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// IntegrationApplication (https://discord.com/developers/docs/resources/guild#integration-application-object)
+type IntegrationApplication struct {
+	ID          Snowflake `json:"id"`
+	Name        string    `json:"name"`
+	Icon        string    `json:"icon"`
+	Description string    `json:"description"`
+	Summary     string    `json:"summary"`
+	Bot         *bool     `json:"bot"`
+}
+
+// Integration (https://discord.com/developers/docs/resources/guild#integration-object)
+type Integration struct {
+	ID                Snowflake               `json:"id"`
+	Name              string                  `json:"name"`
+	Type              string                  `json:"type"`
+	Enabled           bool                    `json:"enabled"`
+	Syncing           *bool                   `json:"syncing"`
+	RoleID            *bool                   `json:"role_id"`
+	EnableEmoticons   *bool                   `json:"enable_emoticons"`
+	ExpireBehavior    *uint8                  `json:"expire_behavior"`
```

Comment:
It's fine, whatever you prefer. I just used a `uint8` to make the types stricter.

Reply:
I think for now it would be better to just keep int as a general value? If we become too restrictive with the types across the library, it's possible that Discord adds a new kind of thing with a higher numeric value that exceeds the limit and causes panics/incorrect values

Reply:
I mean, if Discord updates the type of value for something, Disgo should reflect those changes. Not that it would make that much of a difference if we swapped out an unsigned 8-bit integer with just a regular integer, but it's worth considering since it makes the types as strict as possible and it'd be slightly more efficient for memory allocation I suppose.
@TopiSenpai What do you think?

Reply:
Changing a type mid-release is a breaking change, we can't really do that easily. Additionally my point was that discord changing these values before the library is updated can cause errors while people are on lower versions

Reply:
I'd say we stick with int int64 where it's needed for following reasons you can read about here https://stackoverflow.com/questions/63105394/in-go-when-should-you-use-uint-vs-int